README.md

```diff
@@ -54,41 +54,44 @@ writes are ByteBuffers only which means they are byte arrays or chunks of data.
     final MergedByteBuffers mbb = new MergedByteBuffers();
     
     //This makes a connection to the specified host/port
-    //As the connection is made at this point no reads or writes will be 
-    //pulled off the socket yet.
-    final TCPClient client = new TCPClient("www.google.com", 80);
+    //The connection is not yet actually made, connect() must be called to do that.
+    final TCPClient client = TSE.createTCPClient("www.google.com", 80);
     
     //Here we set the Reader call back.  Everytime the client gets a read
     //This will be executed.  This will happen in a single threaded way per client.
-    //Because it returns the client this is for you could have 1 Reader for many clients
+    //Because it returns the client that the read happened on you can have 1 Reader for many clients
     //assuming you are watching threading between then on your own.
     client.setReader(new Reader() {
         @Override
         public void onRead(Client returnedClient) {
             mbb.add(returnedClient.getRead());
             System.out.println(mbb.getAsString(mbb.remaining()));
+            if(!onReadFuture.isDone()){
```

Comment:
This is not thread safe, you should just construct the SettableListenableFuture with a "false" and set the result without checking if it's done.